File: LeetCode/python3/167.py
class Solution:
    def twoSum(self, numbers: 'List[int]', target: 'int') -> 'List[int]':
        i, j = 0, len(numbers) - 1
        while i < j:
            s = numbers[i] + numbers[j]
            if s < target:
                i += 1
            elif s > target:
                j -= 1
            else:
                return [i + 1, j + 1]
        return []

File: LeetCode/cpp/167.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> ret;
        int i = 0, j = numbers.size() - 1;
        while (i < j) {
            if (numbers[i] + numbers[j] == target)
                break;
            else if (numbers[i] + numbers[j] > target)
                --j;
            else
                ++i;
        }
        ret.push_back(i + 1);
        ret.push_back(j + 1);
        return ret;
    }
};
